.github/workflows/cmake.yml

```diff
@@ -546,21 +546,38 @@ jobs:
             codecov: win64_gcc_compat_no_opt
 
           - name: macOS Clang ASAN
-            os: macos-latest
+            os: macos-13
             compiler: clang
             cxx-compiler: clang++
             cmake-args: -DWITH_SANITIZER=Address
             codecov: macos_clang
 
-          - name: macOS GCC UBSAN
+          - name: macOS Clang ASAN (ARM64)
             os: macos-latest
+            compiler: clang
+            cxx-compiler: clang++
+            cmake-args: -DWITH_SANITIZER=Address
+            codecov: macos_clang_arm64
+
+          - name: macOS GCC UBSAN
+            os: macos-13
             compiler: gcc-10
             cxx-compiler: g++-10
+            ldflags: -ld_classic
```

Comment:
Why `-ld_classic`?

Reply:
Xcode 15 uses a new linker that is not compatible with GCC.
``-ld_classic`` switches to the old linker.

Homebrew ``gcc@11`` and later have a built-in workaround for this issue:

https://github.com/Homebrew/homebrew-core/blob/c0218d50084e300cba26da84028acfd4917ce623/Formula/g/gcc%4011.rb#L92-L96

Reply:
PR updated.